internal/resourcemodifiers/resource_modifiers_validator.go

```diff
@@ -0,0 +1,55 @@
+package resourcemodifiers
+
+import (
+	"strings"
+
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+func (r *ResourceModifierRule) Validate() error {
+	if err := r.Conditions.Validate(); err != nil {
+		return errors.WithStack(err)
+	}
+	for _, patch := range r.Patches {
+		if err := patch.Validate(); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+	return nil
+}
+
+func (p *ResourceModifiers) Validate() error {
+	if !strings.EqualFold(p.Version, ResourceModifierSupportedVersionV1) {
+		return fmt.Errorf("unsupported resource modifier version %s", p.Version)
+	}
+	if len(p.ResourceModifierRules) == 0 {
+		return fmt.Errorf("resource modifier rules cannot be empty")
+	}
+	for _, rule := range p.ResourceModifierRules {
+		if err := rule.Validate(); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+
+	return nil
+}
+
+func (p *JSONPatch) Validate() error {
+	// TODO validate allowed operation
```

Comment:
Will the `TODO` item be implemented in this PR or not?

Reply:
updated.